File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    isValidBST(root) {
        if(!root){
            return true
        }
        function DFS(root,min,max){
            if(root===null)
            return true;
            if(root.val<=min || root.val>=max)
            return false;

            return (
            DFS(root.left,min,root.val)&&
            DFS(root.right,root.val,max)
            )
        }
        return DFS(root,-Infinity,Infinity)
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_set = set()
        left = 0
        best = 0

        for right in range(len(s)):
            while s[right] in char_set:
                char_set.remove(s[left])
                left += 1
            char_set.add(s[right])
            best = max(best, right - left + 1)

        return best

        